node/src/transaction_pool/candidate/transaction_pool_candidate_actions.rs

```diff
@@ -37,12 +38,19 @@ pub enum TransactionPoolCandidateAction {
         peer_id: PeerId,
         transaction: TransactionWithHash,
     },
+    /// Callback for transactions received over pubsub
+    Libp2pTransactionsReceived {
+        peer_id: PeerId,
+        transactions: Vec<TransactionWithHash>,
+        message_id: P2pNetworkPubsubMessageCacheId,
+    },
     #[action_event(level = trace)]
     VerifyNext,
     VerifyPending {
         peer_id: PeerId,
         transaction_hashes: Vec<TransactionHash>,
         verify_id: (),
+        from_source: Option<P2pNetworkPubsubMessageCacheId>,
```

Comment:
Somethig is off, here and on success the field is called `from_source` but only pubsub is valid, it is not a `TransactionPoolMessageSource` like in other places.

Reply:
`get_batch_to_verify` should return the source